stacker/exceptions.py

```diff
@@ -217,3 +217,17 @@ def __init__(self, failed_stacks, *args, **kwargs):
         message = "The following stacks failed: %s\n" % (stack_names,)
 
         super(PlanFailed, self).__init__(message, *args, **kwargs)
+
+
+class GraphError(Exception):
+    """Raised when the graph is invalid (e.g. acyclic dependencies)
+    """
+
+    def __init__(self, exception, stack=None, dependency=None):
+        self.stack = stack
+        self.dependency = dependency
+        self.exception = exception
+        message = ("Error detected when adding '%s' "
```

Comment:
This message will be broken if stack  & dependency aren't set.

Reply:
Where this is used, those things are never not present.

Reply:
Cool - you should make them required then, so we don't hit an edge case.